src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -643,6 +635,27 @@ public void RecordAssertion(AssertionStatus status, string message)
             RecordAssertion(status, message, null);
         }
 
+
+        /// <summary>
+        /// Creates a failure message incorporating failures
+        /// from a Multiple Assert block for use by runners
+        /// that don't know about AssertionResults.
+        /// </summary>
+        /// <returns>Message as a string</returns>
+        public string CreateLegacyFailureMessage()
+        {
+            var writer = new StringWriter();
+
+            if (AssertionResults.Count > 1)
+                writer.WriteLine("Multiple failures or warnings in test:");
```

Comment:
Looks like we are changing the message here. Is it really related to these changes? I'd suggest not performing such a change as part of this PR.